src/components/ActionButton/ToggleActionButton.ts

```diff
@@ -12,6 +12,44 @@ export interface IToggleActionButtonOptions {
 }
 
 export class ToggleActionButton extends Component {
+    /**
+     * Create the deactivated state for a given ToggleActionButton
+     * @param button {ToggleActionButton}
+     */
+    static generateDeactivatedStateInstance(button: ToggleActionButton): IStatefulActionButtonState {
+        return {
+            icon: button.options.activateIcon,
+            tooltip: button.options.activateTooltip,
+            click: () => button.onClick(),
+        };
+    }
+
+    /**
+     * Create the activated state for a given ToggleActionButton
+     * @param button {ToggleActionButton}
+     */
+    static generateActivatedStateInstance(button: ToggleActionButton): IStatefulActionButtonState {
+        return {
+            onStateEntry: function () {
+                this.element.classList.add(ToggleActionButton.ACTIVATED_CLASS_NAME);
+                this.element.setAttribute('aria-pressed', 'true');
+                if (button.options.activate) {
+                    button.options.activate.apply(button);
```

Comment:
You can use the look ahead operator here.
`button.options.activate?.apply(this);`